### ISSUE TITLE
Release KF 1.1

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#5022]
/kind process

We need to identify who will be driving the 1.1 release.  These folks should then

* Identify timelines for the release
   * e.g. cutoff dates for branch cuts
   * Target dates for RCs

This would probably be a good opportunity to update some of the processes and policies around releases.
https://github.com/kubeflow/kubeflow/blob/master/docs_dev/releasing.md

Area | release czar | Tracking Issue
 --- | --- | ---
aws | @Jeffwan  | ~~#5057~~ | 
centraldashboard | | ~~#5068~~
docs | | kubeflow/website#1984
fairing | @jinchihe  | ~~kubeflow/fairing#503~~ |
feast | @woop | |
gcp | @jlewi | kubeflow/gcp-blueprints#46 |
katib | @andreyvelich  | kubeflow/katib#1211 |
kfctl | @krishnadurai, @crobby   | kubeflow/kfctl#352 |
kfserving | @yuzisun @animeshsingh | kubeflow/kfserving#648 |
manifests | @krishnadurai | kubeflow/manifests#1252
metadata | |
minikf | @vkoukis | |
multiuser | @yanniszark  @bmorphism  | #5067, #5068
notebooks | @kimwnasptd , @jtfogarty | #5060, #5068 | 
pipelines | @Bobgy | kubeflow/pipelines#3961 | 
training |  @johnugeorge @andreyvelich @Jeffwan | kubeflow/common#97

Target Dates

* Branch cut June 19

**Comment by terrytangyuan** [comment]
I am happy to help out with this release and get more familiar with the process.

**Comment by jlewi** [comment]
@terrytangyuan thanks do you want to start by taking a look at the release doc and figuring out what issues need to be resolved e.g.

* What permissions need to be granted in order to allow someone to cut releases?
   * (Looks like you might have already sent me a PR).

**Comment by jlewi** [comment]
@terrytangyuan @Jeffwan any updates on process and timelines for 1.1?

One piece of process that we could get started on is ensuring the manifests in kubeflow/manifests have up to date docker images. We should have automation in place to automatically create PRs e.g
https://github.com/kubeflow/manifests/pulls?q=is%3Apr+is%3Aopen+%5Bauto+PR%5D

So we just need to work on getting those PRs merged.

Related to that we need to update our release scripts kubeflow/testing#631 to work with the new manifests.

**Comment by terrytangyuan** [comment]
@jlewi Unfortunately I've been a little overwhelmed with work and family recently. I don't think I'll have too much bandwidth and will just chime in and get familiar with the overall process this time. Please cc me in any PRs/issues that need my reviews or feedback.

**Comment by jlewi** [comment]
@gaocegege @johnugeorge @terrytangyuan @andreyvelich Can someone volunteer to be the release czar of the training related components? 

The release czar would be responsible for coordinating delivery of updated manifests for the following applications

* Katib
* TFJob
* PyTorch
* MxNet?

**Comment by jlewi** [comment]
@kimwnasptd Are you willing to take ownership of coordinating updated manifests for the jupyter web app?

**Comment by jlewi** [comment]
@animeshsingh @cliveseldon @ellistarn @yuzisun Is someone willing to coordinate the delivery of 1.1 manifests and binaries for KFServing?

**Comment by jlewi** [comment]
@rmgogogo @Bobgy Who is going to be the primary person coordinating delivery of KFP for KF 1.1?

**Comment by cliveseldon** [comment]
will by default the latest Seldon manifest from master be brought in or is a cherrypicking still required? @jlewi

**Comment by johnugeorge** [comment]
The task would be releasing releases for individual components and updating it in the manifests.  
right?

> @gaocegege @johnugeorge @terrytangyuan @andreyvelich Can someone volunteer to be the release czar of the training related components?
> 
> The release czar would be responsible for coordinating delivery of updated manifests for the following applications
> 
> * Katib
> * TFJob
> * PyTorch
> * MxNet?

**Comment by Bobgy** [comment]
> @rmgogogo @Bobgy Who is going to be the primary person coordinating delivery of KFP for KF 1.1?

I'll be the primary person coordinating KFP integration with KF 1.1.

**Comment by jlewi** [comment]
@johnugeorge Here's a quick list of tasks for individual release czars

1. Make sure up to date docker images are published
1. Make sure up to date manifests are in kubeflow/manifests
1. Make sure docs get updated

@cliveseldon We will cut a KF 1.1. branch from kubeflow/manifests master at some point. So whatever is on master at that point will become the starting point for what's in 1.1. If you make additional changes after that the release branch is cut changes would need to be cherry-picked onto the release branch.

**Comment by johnugeorge** [comment]
I will work with @andreyvelich for release of training components(TFJob and PyTorchJob) and Katib
/cc @andreyvelich

**Comment by terrytangyuan** [comment]
Awesome. Thanks! @johnugeorge @andreyvelich

**Comment by yuzisun** [comment]
> @animeshsingh @cliveseldon @ellistarn @yuzisun Is someone willing to coordinate the delivery of 1.1 manifests and binaries for KFServing?

I will coordinate the KF 1.1 manifest release for KFServing.

**Comment by animeshsingh** [comment]
add me along KFServing as well with @yuzisun 

For the most part KFServing should be ready to go with release 0.3 (unless we plan to cut another one- which I wouldn't recommend unless necessity) as most of the heavy lifting to upgrade to KFServing 0.3 and Kustomize v3 manifests was done and tested last week in this PR here https://github.com/kubeflow/manifests/pull/1156

We will need to run through notebooks et all to ensure there are not rough edges

**Comment by Bobgy** [comment]
### Pipelines Area Status
(Reporting status here, because the meeting is not China time zone friendly.)
Major work items left:
1. Complete KFP 1.0 release (rc1 just cut today)
2. Integrate KFP 1.0 into KF 1.1 manifest (argo DONE, others to be worked on)
3. Enable KFP multi user mode

In total, I'm roughly estimating 2 weeks needed, but there's a lot of details there that might turn out to be harder.

#### How to report status
I'll create a kanban project in kubeflow/pipelines repo.

#### What items are in jeopardy
If I don't have enough bandwidth, multi user mode might need to be postponed.

#### Help I need
Code review and discussion on above work items. I haven't found a good way to organize some manifests for KFP multi user mode yet.

Also I need to review the plan with my manager.

**Comment by jlewi** [comment]
@jinchihe Are you willing to drive the fairing release?

**Comment by vkoukis** [comment]
A status update: We'll produce a MiniKF release after KF 1.1 is out, this is what we have been doing so far, and make sure everything works e2e.

**Comment by woop** [comment]
@jlewi I had a chat with @jbottum about getting involved as an area lead for Feast. That doesn't exist as an area yet, but I can drive that if you are open to it.

**Comment by jlewi** [comment]
@woop That sounds great; it doesn't look like we have any docs on www.kubeflow.org about Feast. That might be the biggest item. I think its perfectly fine to link from www.kubeflow.org to existing Feast docs or markdown in GitHub.  Filed: kubeflow/website#1981 anything else that should be done for KF 1.1?

**Comment by woop** [comment]
Thanks @jlewi 
> anything else that should be done for KF 1.1?
Let's start with the documentation.

**Comment by Jeffwan** [comment]
@jlewi 

I was off last week and just come back. 

I will help on training operator release as well. I've finished kubeflow/common migration for most of the operators. I think we will get these done by v1.1. 

AWS side, we still have a few things not done yet. We will try to make it by release date.

**Comment by jinchihe** [comment]
> @jinchihe Are you willing to drive the fairing release?

@jlewi Yeah, Thanks.

**Comment by jlewi** [comment]
@adrian555 @animeshsingh @kkasravi @yanniszark We still need an owner for kfctl; is one of you willing to volunteer? You would be responsible for cutting the kfctl branch and pushing releases.

@swiftdiaries @terrytangyuan @yanniszark we still need an owner for kubeflow/manifests. I think the responsibility would be to cut the release branch and tag it. Do I have a volunteer?

In addition, it looks like we are still missing owners for the following areas:

* notebooks
* website

It looks like to create releases you will need [write permission](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization) on the repository.

Area owners can you please create appropriate GitHub teams related to your repo/project/WG and grant write permissions to the requisite repositories so you can create releases. This should be done in
https://github.com/kubeflow/internal-acls/blob/master/github-orgs/kubeflow/org.yaml

**Comment by jlewi** [comment]
Area leads could you please do the following

1. File an issue specific to the release of your area/application 
1. Create appropriate GitHub teams in [org.yaml](https://github.com/kubeflow/internal-acls/blob/master/github-orgs/kubeflow/org.yaml) with necessary repository permissions to cut release branches and artifacts
    * Please name these teams "${foo}-release" 
    * Grant only those individuals who are participating in the release membership
    * Only grant access to the repos you will actually be cutting RCs for


/cc @johnugeorge @andreyvelich @Bobgy @yuzisun @animeshsingh @jinchihe

**Comment by jlewi** [comment]
A couple other areas where we still need owners

* metadata
* centraldashboard
* multiuser
   * @yanniszark any interest?

**Comment by Jeffwan** [comment]
@jlewi 

I am familiar with kfctl, manifest, profile-controller,  notebook controllers. If you still need release owners in these area, Let me know.

**Comment by krishnadurai** [comment]
@jlewi I should be able help in releasing Manifests and Kfctl - having known the processes since last 2 releases.

**Comment by jlewi** [comment]
@yanniszark @Jeffwan You both indicated an interest in helping out with multi-user. Barton also chimed in on slack but I don't know his GitHub id. Can one of you please file an issue to track the release of those components and then link to it here?

@krishnadurai and @vpavlin Can one of you file an issue for manifests, link it here, and then use that issue to coordinate?

**Comment by jinchihe** [comment]
The Fairing release tracking Issue: https://github.com/kubeflow/fairing/issues/503

**Comment by Bobgy** [comment]
Pipelines release tracking issue: https://github.com/kubeflow/pipelines/issues/3961
and release tracking project: https://github.com/kubeflow/pipelines/projects/5

**Comment by andreyvelich** [comment]
For Katib all docs and manifests have been updated (tracking issue https://github.com/kubeflow/katib/issues/1211).
The new release has been published: https://github.com/kubeflow/katib/releases/tag/v0.9.0.

**Comment by Jeffwan** [comment]
https://github.com/kubeflow/common/issues/97 training operators tracking issue.

https://github.com/kubeflow/kubeflow/issues/5057 AWS tracking issue.

**Comment by animeshsingh** [comment]
KFServing will be tracked via this cc @yuzisun 
https://github.com/kubeflow/kfserving/issues/648

**Comment by yuzisun** [comment]
We found a [KFServing issue](https://github.com/kubeflow/kfserving/issues/874) with Istio/Dex setup where inference service is getting RBAC access denied even without istio sidecar injected, the issue is that the istio local cluster gateway gets blocked because `istio-system` is not whitelisted. We found in [istio/dex kfdef](https://github.com/kubeflow/manifests/blob/v1.0-branch/kfdef/kfctl_istio_dex.v1.0.2.yaml#L45), it is using mode `ON` instead of `ON_WITH_EXCLUSION` despite `istio-system` is whitelisted [here](https://github.com/kubeflow/manifests/blob/master/istio/istio/base/kf-istio-resources.yaml#L113), I have checked gcp is actually doing `ON_WITH_EXCLUSION` [here](https://github.com/kubeflow/manifests/blob/5de67258ca92a59f29d025d69a566310fce208f5/istio/iap-gateway/base/istio-ingressgateway.yaml#L59),  this sounds like an existing bug in manifests, can someone help confirm?

**Comment by adrian555** [comment]
@yuzisun were you referring to the issue we hit last time we upgraded the kfserving to v0.3.0?

**Comment by yuzisun** [comment]
> @yuzisun were you referring to the issue we hit last time we upgraded the kfserving to v0.3.0?

yes that's correct, someone else also reported the [issue](https://github.com/kubeflow/kfserving/issues/874) here, @adrian555 can you help verify my [fix](https://github.com/kubeflow/manifests/pull/1257)?

**Comment by adrian555** [comment]
@yuzisun yes, I can run it again. But last time that was the work around I made it to work.

**Comment by jlewi** [comment]
@adrian555 @yuzisun Can we move the discussion of that KFServing;Dex;Istio into a separate issue?

**Comment by animeshsingh** [comment]
Move it here kubeflow/kfserving#648 for the overall release issues for KFS

**Comment by jlewi** [comment]
Please join #release in slack for more real time updates of the release.

* the apps cd pipelines have been fixed
* application owners please review PRs to update your apps and ensure they get merged
    * https://github.com/kubeflow/manifests/pulls/kubeflow-bot

**Comment by jlewi** [comment]
I think we will cut 1.1 branch of manifests and kubeflow today. Any objections?

**Comment by woop** [comment]
@jlewi documentation from my side will be ready by tomorrow morning. Your cut doesn't include the website right?

**Comment by jlewi** [comment]
@woop no website won't be finalized for a couple weeks.

**Comment by jlewi** [comment]
Update:

* www.kubeflow.org has been pointed at the v1-0-branch
   * So we can begin updating master to v1.1 
* kubeflow/kubeflow 1.1 branch has been cut
* kubeflow/manifests 1.1 branch hasn't been cut yet
   * Waiting for kubeflow/manifests#1287 to merge

**Comment by jlewi** [comment]
kubeflow/manifests 1.1 branch has been cut any future changes to manifests will need to be cherry-picked onto the release branch.

Application owners

* Now that kubeflow/manifests 1.1 branch has been cut you can update those images to use 1.1 tagged images
* If you are using apps-cd PRs should automatically be opened up

Platform owners

* Please make any changes necessary on the 1.1. branch to prepare your platform for release
* At this point please begin doing E2E testing to verify the 1.1 release is working on your platform

Docs
* We should be updating the docs on kubeflow/website@master in preparation for the 1.1 launch

**Comment by Jeffwan** [comment]
AWS has some minor changes not merged yet and everything should be ready by this week. I will make sure they are back ported to release branch

**Comment by Jeffwan** [comment]
AWS side is clear and we don't have pending changes.

**Comment by 8bitmp3** [comment]
@jlewi with your guidance i can run `/docs` for KF v1.1. if it's available 👍

**Comment by jlewi** [comment]
How's everyone's release qualification going?

On the GCP side we've been testing the v1.1.0-rc.0 manifests and have encountered a couple issues

* kubeflow/gcp-blueprints#74 knative and kfserving is missing from our deployment
* the other [test failure](https://k8s-testgrid.appspot.com/sig-big-data#kubeflow-gcp-blueprints-periodic-1-1&group-by-hierarchy-pattern=%5B%5Cw-%5D%2B) appears to be a bug in our tests
* mnist is passing
* We would like to get xgboost integrated into our test suite kubeflow/gcp-blueprints#78 as that provides additional coverage (e.g. pipelines)

It looks like we don't have much in the way of qualification for KFServing kubeflow/kfserving#930

**Comment by Bobgy** [comment]
FYI, I'm using https://github.com/kubeflow/pipelines/issues/3241 as tracker issue for KFP multi user mode.

**Comment by jlewi** [comment]
On the GCP side we are waiting on two bug fixes before cutting the next RC of manifests

This is being tracked in kubeflow/manifests/issues/1252

The fixes we need are 

1. Fixes to install knative and kfserving on GCP
1. KFP multi-user cherrypicks

@animeshsingh @vpavlin @jinchihe @yanniszark @johnugeorge @Jeffwan How is your release qualification going what are you doing to qualify your release?

**Comment by jinchihe** [comment]
For Fairing, is that acceptance if the mnist example passed? For KFServing, I'm going to add e2e for that. Thanks.

**Comment by jlewi** [comment]
> For Fairing, is that acceptance if the mnist example passed? 

@jinchihe for now yes.

**Comment by jlewi** [comment]
The GCP tests are passing pretty reliably with the branch:
https://github.com/kubeflow/manifests/tree/v1.1-rc.1

I just created a new RC to pick up some centraldashboard changes
https://github.com/kubeflow/manifests/releases/tag/v1.1-rc.2

and more importantly because v1-1-rc.1 is a branch and not a tag.

**Comment by jlewi** [comment]
@Bobgy everything looks good to me I'm going to finalize v1.1.0

**Comment by PatrickXYS** [comment]
@jlewi 

On AWS side:
1. Multi-user KFP adoption
2. Migrate to manifest V3.

Has been tested + merged + cherry-picked into v1.1 branch. I think AWS is good to go with v1.1.0

Ref: https://github.com/kubeflow/manifests/pull/1410

**Comment by adrian555** [comment]
@jlewi we are testing the multi-user kfp on IBM platforms.

**Comment by jlewi** [comment]
I think at this point we mostly just need to finalize the docs in order to finalize the release.
https://github.com/kubeflow/website/issues/1984

/cc @8bitmp3 @RFMVasconcelos

**Comment by animeshsingh** [comment]
@jlewi i believe the most demanded feature for  KF1.1 is multi-user KFP v1.0  - Since this came out very recently, we are in the last legs of finalizing the manifests et all for IBM Cloud. I believe there will be others who would want to finalize multi-user functionality at their end?

**Comment by animeshsingh** [comment]
@jlewi please look at the list of issues being reported on multi-user pipelines here
https://github.com/kubeflow/manifests/issues/1364#issuecomment-665218071

**Comment by Jeffwan** [comment]
AWS meets some issues in multi-user Pipeline as well https://github.com/kubeflow/manifests/issues/1430
Need extra days to fix these issues.

**Comment by jlewi** [comment]
@animeshsingh @Jeffwan 1.1.0 was finalized. When your fixes are on the 1.1 branch we can look at cutting a 1.1.X.

**Comment by Jeffwan** [comment]
@jlewi Great. some of our v3 manifests changes and multi-user changes are not included in v1.1.0 tag. We plan to use v1.1-branch at this moment in kfdef manifest. Once we cut patch release v1.1.x, we will switch over. 

For multi-user pipeline, currently, only pending issue is https://github.com/kubeflow/pipelines/pull/4285. 

Our manifest and doc will be ready by EOD. KFP python sdk can be installed separately in notebook later.

**Comment by jbottum** [comment]
@jlewi Are we ready to cut over the web site to 1.1 ?

**Comment by jlewi** [comment]
@jbottum please kubeflow/website#1984

**Comment by jlewi** [comment]
@jbottum docs have been published. Some of our docs are still outdated (e.g. they have an outdated banner on them indicating they need to be refresh).

I think we can go ahead and publish the 1.1 blog post.

**Comment by animeshsingh** [comment]
https://github.com/kubeflow/website/pull/2106
this one just went in @jlewi

**Comment by sylus** [comment]
@animeshsingh i am a bit confused it was said in slack that Kubeflow 1.1 is released and you can see it in releases for manifests but the project board doesn't reflect any of this.

Is Kubeflow 1.1 officially released or not?

**Comment by jlewi** [comment]
1.1 has been released
